mono/metadata/icall.c

```diff
@@ -2797,6 +2802,62 @@ ves_icall_reflection_get_token (MonoObjectHandle obj, MonoError *error)
 	return mono_reflection_get_token_checked (obj, error);
 }
 
+ICALL_EXPORT guint32
+ves_icall_reflection_get_token1 (MonoObjectHandle obj, MonoError *error)
```

Comment:
Why do we need all this function duplication?

Reply:
The native function is exposed with multiple managed names, and each managed name produces a wrapper. There are duplicate wrappers, one per managed name, with name based on the native name. An additional feature of the macros might be to reuse a wrapper with a specified name.

Reply:
Multiple icalls have the same underlying native implementation.
The wrapper is named based on that implementation.

`System.Arbitrary1.foo` => `ves_foo_raw` => `ves_foo`
`System.Arbitrary2.foo` => `ves_foo_raw` => `ves_foo`

but that produces two (identical) ves_foo_raws.
I could probably alter their names based on __LINE__ or, the enum that is the first parameter to ICALL(), or give the ICALL macro an ability to reuse another icall's wrapper.

Reply:
Or declare them no `NOHANDLES()` but and call the `_raw`. That is slightly abstraction violating but might be pretty good otherwise.

Reply:
That won't work -- I confused raw pointers and raw handles.

Reply:
I guess HANDLES_NO_WRAPPER.
marshal-ilgen is still going to produce duplicates.
If/when we move to the marshal-ilgen-less solution though, it works.